### PR TITLE
FEATURE: Priority Groups for group tracked posts

### PR DESCRIPTION
**Pull request by jdmartinez1062**
**Problem**
The customer wants to be able to change the group that is tracked in a given topic if a `priority group` comments on said topic. Also, the customer wants to change the `title` displayed when a user hovers on the icon of the currently tracked group for a topic. Currently, we are using the `name` of the group that is being tracked in a topic, however, the customer wants to use the `full_name`

**Solution**
Change the used attribute to the `full_name` instead of the `name` of the group. Create a new method with a modified SQL that allows to change of the tracked group for a given topic

### CODE REVIEW

**Review by featheredtoast — CHANGES_REQUESTED** [review]
Hey Juan, thanks for this!

Is there a reason for the site setting to allow groups to become priority? Seems like just an extra button to tick to turn it on, might as well just have groups opt into being priority groups; as if a site doesn't pick any, then none exist.

Let's update the copy of `track_priority_groups` option to be something like
"Display this group's icon in the topic list even when another tracked group user has replied first."

I'd also prefer the `track_priority_group ` setting itself to be something like `track_posts_with_priority`, since the word "group" is already implied, and to be inline with the `track_posts` name already present

Let's also fix the linting and get those tests up and running before merging

**Review by featheredtoast — CHANGES_REQUESTED** [review]
@jdmartinez1062 take a look at changes, there's still some suggested cleanup for you. Commit has majority of the comments here.

TLDR, inside the big query, have two queries: one that queries priority groups, and another that queries lower priority groups and union. It's more reliable to be able to recalculate the entire tracking state for posts.

We're also missing a test here, pretty sure you were missing what happens with priority tracking if admin group controller's `update_track_posts` gets called. Previously it looks like it'd overwrite all states with the default query, how we had it before.

Have a look to see if you understand the new query here, and let me know if it all makes sense.

**Review by featheredtoast — APPROVED** [review]

**Review by featheredtoast — CHANGES_REQUESTED** [review]
Hmm the query may still need some tweaking, the topic category still needs to jump to the "first" tracked post.

**Comment by featheredtoast** [comment]
Actually after considering it, it kinda makes sense, if priority it'd bring them to the priority group's post, rather than the "first tracked".

**Review by featheredtoast — APPROVED** [review]
